Lib/unittest/mock.py

```diff
@@ -648,9 +648,18 @@ def reset_mock(self,  visited=None,*, return_value=False, side_effect=False):
         if side_effect:
             self._mock_side_effect = None
 
-        for child in self._mock_children.values():
+        for key, child in self._mock_children.items():
             if isinstance(child, _SpecState) or child is _deleted:
                 continue
+            if (
+                return_value
+                and issubclass(type(child), MagicMock)
+                and _is_magic(key)
+            ):
+                # Don't reset return values for magic methods,
+                # otherwise `m.__str__` will start
+                # to return `MagicMock` instances, instead of `str` instances.
+                return_value = False
```

Comment:
It kinda feels like this logic should be the responsibility of the `reset_mock` implementation in `MagicMock`, could the logic be moved there?

Reply:
I will try, thanks for the idea! 👍